lib/rubygems/user_interaction.rb

```diff
@@ -575,6 +575,57 @@ def done
   end
 
   ##
+  # A progress reporter that behaves nicely with threaded downloading.
+
+  class ThreadedDownloadReporter
```

Comment:
Hm. I'm not sure how the default progress reporter will play with what we're doing; I suspect it'll look like a hot mess (hence why this class was created to start with). I also don't remember how the various `DownloadReporter` things tie into everything else. Will have to look into this in more detail.